falcon/request.py

```diff
@@ -2334,6 +2334,90 @@ def get_param_as_json(
 
         return val
 
+    def get_param_as_dict(
+        self,
+        name: str,
+        required: bool = False,
+        deep_object: bool = False,
+        store: StoreArg = None,
+        default: Any | None = None,
+    ) -> Any:
+        """Retrieve a query parameter as a dictionary.
+
+        Supports OpenAPI's deep object style (`param[key]=value`)
+        and list-based key/value pairs (e.g., `param=a,b,c,d`).
```

Comment:
```suggestion
        and list-based key/value pairs (e.g., `param=k1,v1,k2,v2`).
```

just to make it clearer